### PR TITLE
Add `--build-before-test` to run_tests.py

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
For running the tests locally.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: testing/run_tests.py [hunk 1]
```diff
@@ -281,36 +281,105 @@ def RunDartTest(build_dir, test_packages, dart_file, verbose_dart_snapshot, mult
                       forbidden_output=forbidden_output, expect_failure=expect_failure)
 
 
-def EnsureDebugUnoptSkyPackagesAreBuilt():
+def EnsureDebugUnoptSkyPackagesAreBuilt(check_only):
```
Comment (dnfield) [inline testing/run_tests.py:284]: nit: here and elsewhere, change `check_only` to something like `do_build`.

It's slightly confusing right now to track how this gets set as an inversion of a top level argument to the script.

**Review by dnfield — APPROVED** [review]
LGTM with nit

**Review by zanderso — CHANGES_REQUESTED** [review]
Since we're running this script on CI, I think it's safer to avoid any chance that we might use it to trigger a build by just not having code to do so in it at all. What's wrong with just following the instructions in the error messages?

**Comment by LongCatIsLooong** [comment]
@zanderso in the `README.md` files people were told to run the script locally after they add new tests. There may not be error messages if people have built the test targets before and added more tests (It took me a while to realize my new tests were actually not running since the ios test target was not rebuilt). 
I can change the readme files to add the build instructions, or create a different script for running the tests locally, or assert `LUCI_CI` is not set when we invoke gn/ninja. Which one makes the most sense?

**Comment by zanderso** [comment]
My suggestion would be to update the `README.md`. Folks should be in the habit of triggering a build explicitly after making source changes.

**Comment by chinmaygarde** [comment]
Closing per Zachs comment.

**Pull request closed without merging**